src/aspire/image/image.py

```diff
@@ -416,6 +416,89 @@ def shift(self, shifts):
 
         return self._im_translate(shifts)
 
+    def legacy_whiten(self, psd, delta):
+        """
+        Apply the legacy MATLAB whitening transformation.
+
+        Note that this legacy method will compute the convolution in
+        (complex)double precision regardless of this instances
+        `dtype`.  However, the resulting image stack will be cast to
+        the instance `dtype`.
+
+        :param psd: PSD as computed by `LegacyNoiseEstimator`.
+        :param delta: Threshold used to determine which frequencies to whiten
+            and which to set to zero. By default all `sqrt(psd)` values
+            less than `delta` are zeroed out in the whitening filter.
+        """
+        n = self.n_images
+        L = self.resolution
+        j = L // 2
```

Comment:
`L_half`?

Reply:
Yeah, I get it, but it made the slices later really long.  I ended up needing refactoring it to be readable.